src/directions/main.ts

```diff
@@ -755,7 +759,11 @@ export default class MapLibreGlDirections extends MapLibreGlDirectionsEvented {
   protected assignWaypointsCategories() {
     this._waypoints.forEach((waypoint, index) => {
       const category = index === 0 ? "ORIGIN" : index === this._waypoints.length - 1 ? "DESTINATION" : undefined;
-      if (waypoint.properties) waypoint.properties.category = category;
+
+      if (waypoint.properties) {
+        waypoint.properties.index = index;
```

Comment:
Another separate-PR-candidate. Basically, just saves a waypoint's index (for each of the waypoints) in their Feature Properties object (allows to e.g. display number of each waypoint as symbol (text) on the map).